packages/typegen/src/generate/rpc.ts

```diff
@@ -111,9 +113,17 @@ export function generateRpcTypes (importDefinitions: { [importPath: string]: Rec
   });
 }
 
-export function generateDefaultRpc (): void {
+export function generateDefaultRpc (dest = 'packages/api/src/augment/rpc.ts', metadata = staticData, extraTypes: Record<string, Record<string, { types: Record<string, any> }>> = {}): void {
+  const registry = new TypeRegistry();
+
+  registerDefinitions(registry, extraTypes);
+
+  registry.setMetadata(new Metadata(registry, metadata));
```

Comment:
As suggested, no, metadata is not needed as per the original implementation here. RPCs are not communicated over metadata.